Add Your Trips section and commute maps

rachelekm opened the pull request:
## Overview

Implements the neighborhood details "Your trip" section, including:

- Translatable text content for section and edit trips button (currently does not open until trip wizard implemented)
- Implementing map with Destination-neighborhood route geojson
- Add route styling with segments styled according to transit type (brought over from Taui)
- Refactors times/paths fetch state so that multiple destinations' data can be fetched upfront
- Adds accordion base component, for use to display commutes group
- Adds dynamically created Google maps directions link

### Checklist

- [x] `fixup!` commits have been squashed
- [x] `CHANGELOG.md` updated with summary of features or fixes, following
      [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines
- [ ] `README.md` updated if necessary to reflect the changes
- [x] Run `./scripts/format` to lint, format, and fix the application source code.
- [x] CI passes after rebase

### Demo

https://github.com/user-attachments/assets/aba6c6dc-3ad6-4b76-9e8f-4794d6802cf6


<img width="331" height="842" alt="Screenshot 2025-10-02 at 4 55 04 PM" src="https://github.com/user-attachments/assets/97695d86-dd02-4989-847a-9a598e3b1818" />


## Testing Instructions

 * `scripts/server`
 * Open Chrome networks tab and confirm action dispatched for getAllTimesAndPathsData in redux logger
 * Confirm getAllTimesAndPathsData is fulfilled and timesAndRoutesData has a key and data for every destination
 * Select neighborhood
 * Confirm "Your Trips" section of details matches figma designs (the edit trips button will not yet work)
 * Confirm first destination in accordion is highlighted and map route displays in map
 * Click google map directions link and confirm map directions open in new tab
 * Verify route in map with google map route
 * Select other destinations and confirm map appropriately updates with new route, bounds, and directions link
 * Repeat steps for mobile view and confirm matches figma designs
 * Confirm no issues on language change (example translations added to google maps directions link)


Resolves #664

rachelekm committed:
Add Your Trips base UI

rachelekm committed:
Add base route UI for destinations

Brings in transit geojson styling from Taui as well as fixes an error with missing route path segment creation if no route type supplied

rachelekm committed:
Add route anchor points styling

rachelekm committed:
Add dynamic directions link

rachelekm committed:
Refactor to fetch all destinations data upfront

rachelekm committed:
Fix incorrectly cached memoization

rachelekm committed:
Add base Accordion component

rachelekm committed:
Add dynamic commute group

rachelekm commented on src/frontend/src/reducers/networks/utils/fetchAndProcessDataByDestination.ts:
```diff
@@ -0,0 +1,111 @@
+// Refactored from old codebase loadDataset and fetchAllTimesAndPathsForCoordinate at commit 6f17e33
+// New source code copied from latest Taui v3
+// ATTR: Taui by Conveyal, included under the MIT license (https://github.com/conveyal/taui/blob/dev/LICENSE)
```
This file is just the logic from getTimesAndPathsDataForPlace pulled out into a fn that can also be called from the bulk data fetch thunk

rachelekm committed:
Refactor draw route for any destination-->neighborhood

rachelekm commented on src/frontend/src/reducers/neighborhoods/selectors/drawNeighborhoodRoutes.ts:
```diff
@@ -0,0 +1,187 @@
+// Refactored from old codebase at latest commit before introducing listings: 8ca3b94
+// New code copied from latest Taui v3 to support routes with MapLibre
+// ATTR: Taui by Conveyal, included under the MIT license (https://github.com/conveyal/taui/blob/dev/LICENSE)
+
```
This file is essentially the same as old drawNeighborhoodRoute except now it's wrapped in a function to allow passing in dynamic destination and neighborhood identifiers. Previously this selector pulled the active destination and active neighborhood from state, but we need 1) routes for each destination for an active neighborhood in the details view 2)  routes for each destination for every favorited neighborhood in the compare view. This allows creating a destination-->neighborhood route for any given combo

rachelekm commented:
@philippschmitt tagging you here as well for a polish pass since this PR covers the neighborhood details + your trips section (noting that the transit directions in the map corner are being handled in an upcoming issue).

rachelekm committed:
Simplify initial map bounds fit

philippschmitt committed:
Rename tab to 'Find units'

philippschmitt committed:
update icons

philippschmitt committed:
tweak toggle button

philippschmitt committed:
Tweak map: Increase boundary contrast, remove bus stop labels that show up on close zoom levels

philippschmitt committed:
Add shadow to 'Start' map icon and tweak minimap style

philippschmitt committed:
Trips section polish

philippschmitt committed:
Tweak accordion to remove layout jumping

philippschmitt committed:
Make entire image of area clickable

philippschmitt committed:
Misc style tweaks for polish

philippschmitt commented:
@rachelekm Just completed a polish pass for the entire NeighborhoodDetail card — I hope this is how you intended it. This PR looked great already but since there are so many new elements the tweaks do add up. If anything is out of scope let me know!

Hopefully the commits are self-explanatory. I have two additional notes:

1. On desktop, it doesn't make sense that users can de-select trips. Could we disable that? 
![CleanShot 2025-10-07 at 19 19 16](https://github.com/user-attachments/assets/2d9994e5-300f-45d4-ae9c-0bf7b5dde093)

2. Do we need to display the CC license for every image? I'd prefer the format `Photo by {artist}. Source.`
<img width="1722" height="176" alt="10689" src="https://github.com/user-attachments/assets/0c8f1b5b-0acd-4984-a590-649f8f484566" />

rachelekm committed:
Clean up Accordion selection

- Prevents users from deselecting expanded AccordionItem
- Ensures overridePanelOpen works as expected for "car" mode

rachelekm committed:
Update photo license caption

rachelekm commented:
Thank you so much, @philippschmitt! The commits make sense to me and the changes look great! I pushed up two more commits to address your suggestions. Also noting that I rebased with develop to resolve immediate merge conflicts so you will need to force pull this branch, if needed.

> 
> 1. On desktop, it doesn't make sense that users can de-select trips. Could we disable that?

That makes sense and implemented in [89e3e29](https://github.com/azavea/echo-locator/pull/724/commits/89e3e297002101b7f445dc959c87a4ab35e0710e). I also added a little bit of logic cleanup for the "car" version as well in that commit

https://github.com/user-attachments/assets/5a591f52-eba7-4153-89f6-c8aa6a707180



> 2. Do we need to display the CC license for every image? I'd prefer the format `Photo by {artist}. Source.`

Nope, I just added everything we had available in by default! I updated the caption to your suggestion and also added translations in [b57728b](https://github.com/azavea/echo-locator/pull/724/commits/b57728b7fd34513a0cfc9afe23c71e4f4be083c7).
<img width="332" height="305" alt="Screenshot 2025-10-08 at 12 32 27 PM" src="https://github.com/user-attachments/assets/e13db1bf-252d-4fd5-a6a3-92b78afaf3db" />

aaronxsu commented on src/frontend/src/components/YourTrips/TripMap.tsx:
```diff
@@ -0,0 +1,136 @@
+    }, [loaded]);
+
+    useEffect(() => {
+        mapRef.current?.fitBounds(bounds, {
```
In the previous PR, we were discussing if this is necessary. I wonder if this is now needed since bounds can change?

rachelekm replied:
After our discussion I got rid of the other useEffect setting initial bounds with padding, but we'll need to keep this one in to capture the bounds changing in desktop-view since that's a shared map component that isn't re-initialized on destination change. This also shouldn't cause redundancy In mobile view since each destination panel has its own map component and the bounds never change for those.

aaronxsu commented on src/frontend/src/components/YourTrips/TripMap.tsx:
```diff
@@ -0,0 +1,136 @@
+import { useTranslation } from "react-i18next";
+import { Link } from "react-aria-components";
+
+import type { Destination } from "src/reducers/userProfile/types";
```
It could just be `from "reducers/userProfile/types";`

aaronxsu commented on src/frontend/src/components/YourTrips/TripMap.tsx:
```diff
@@ -0,0 +1,136 @@
+    routeEndPointStyle,
+} from "./tripLayerStyles";
+import { yourTripsStyles } from "./YourTrips.styles";
+import { createGoogleDirectionsURL } from "src/libs/getLinkURLs";
```
Could just be `libs/getLinkURLs`
Same applies to the next two imports. I think I see similar issue in other modules/files. I'd examine them and remove the `src` bit

aaronxsu commented on src/frontend/src/components/YourTrips/YourTrips.styles.ts:
```diff
@@ -0,0 +1,35 @@
+import { tv } from "tailwind-variants";
```
It might make more sense to make this file a `.ts` rather than a `.tsx`

aaronxsu commented on src/frontend/src/components/YourTrips/CommuteGroup.tsx:
```diff
@@ -0,0 +1,103 @@
+        isMobile: isMobile,
+    });
+    const [selectedMapDestination, setSelectedMapDestination] =
+        useState<Destination>(trips[0].destination);
```
Is it safe to do `trips[0].destination` here?

aaronxsu commented on src/frontend/src/components/YourTrips/CommuteGroup.tsx:
```diff
@@ -0,0 +1,103 @@
+        <div className={styles.commuteGroupWrapper()}>
+            <Accordion
+                defaultExpandedKeys={["0"]}
+                expandedItemCallback={keys => {
```
I'd not have this inline, but move it into a callback function to be defined above.

aaronxsu commented on src/frontend/src/components/YourTrips/CommuteGroup.tsx:
```diff
@@ -0,0 +1,103 @@
+                        {isMobile && (
+                            <TripMap
+                                start={trip.destination}
+                                end={trips[0].neighborhoodZipcode}
```
Is it safe to do `trips[0].neighborhoodZipcode` here? Same question for line 94

aaronxsu commented on src/frontend/src/components/YourTrips/CommuteGroup.tsx:
```diff
@@ -0,0 +1,103 @@
+                        titleContentRight={
+                            <Range
+                                start={
+                                    allNeighborhoodCommutes[
```
I'd not have them inline, and instead would create a method to get the value for the start and end props

aaronxsu commented on src/frontend/src/components/YourTrips/YourTrips.tsx:
```diff
@@ -0,0 +1,101 @@
+    const favoritedNeighborhoods: string[] = [];
+    const useTransit = useAppSelector(selectUseTransit);
+
+    const tripsByDestination = destinations.reduce(
```
Would it be necessary to wrap it in a `useMemo`?

aaronxsu commented on src/frontend/src/components/YourTrips/YourTrips.tsx:
```diff
@@ -0,0 +1,101 @@
+                tripToNeighborhood: !!tripToNeighborhood,
+            };
+            tripsFromDest.push(trip);
+            return tripsFromDest;
```
I'd combine the above two lines to just `return [...tripsFromDest, trip]`, and maybe even put the entire trip object in the return and save a `{}` block

aaronxsu commented on src/frontend/src/components/YourTrips/YourTrips.tsx:
```diff
@@ -0,0 +1,101 @@
+    });
+
+    const destinations = useAppSelector(selectUserDestinations);
+    const tripToNeighborhood = !!activeNeighborhood;
```
For boolean variables, I usually find defining `isTripToNeighborhood` be more readable.

aaronxsu commented on src/frontend/src/hooks/useGetDestinationToNeighborhoodRoute.tsx:
```diff
@@ -0,0 +1,18 @@
+    );
+
+    const routeGeoJson = useAppSelector(selectDestinationToNeighborhoodRoute);
+    return routeGeoJson;
```
I'd combine the above two lines and return directly.

aaronxsu commented on src/frontend/src/libs/getLinkURLs.ts:
```diff
@@ -9,11 +10,19 @@ const URLsBySite = {
+export const createGoogleDirectionsURL = (
+    neighborhood: string,
+    destination: Destination,
+    tripToNeighborhood: boolean,
```
Same suggestion as above, re: naming a boolean var

rachelekm committed:
Update network mode enum naming

Duplicate type names was unecessary and confusing. Rrefactoring the network mode types with better type names and less redundency will prevent issues when dynamically setting activeMode in state down the road

rachelekm committed:
Fix path/time data fetching too early

aaronxsu commented on src/frontend/src/pages/Discover/Discover.tsx:
```diff
@@ -78,32 +78,33 @@ const Discover = () => {
-                d => d.location.label === activeDestination
-            );
-            destination && dispatch(getTimesAndPathsDataForPlace(destination));
+        if (destinations && !timesAndRoutesData) {
```
Just noting here that @rachelekm and I paired on this part and figured that we need to check neighborhoods and network in this effect as well.

aaronxsu reviewed (commented):
This works great and looks very well! I left some questions in the comments and would like to hear what you think!

rachelekm committed:
Update imports to remove relative path

rachelekm committed:
Update file type

rachelekm committed:
Clean up commute group logic

rachelekm committed:
Clean up Your Trips and detail polish

rachelekm commented:
Thank you, @aaronxsu! I addressed those fixes in the last four commits I pushed up and this is set for another look! Also noting that since the merge conflict is for the components page I'll hold off on addressing that until merge.

aaronxsu commented on src/frontend/src/components/YourTrips/CommuteGroup.tsx:
```diff
@@ -23,34 +25,53 @@ const CommuteGroup = ({
+
+    useEffect(() => {
+        setSelectedMapDestination(initialTrip.destination);
+    }, []);
```
Should we add `initialTrip` as a dep here?

rachelekm replied:
Fixed in latest rebase 👍

aaronxsu commented on src/frontend/src/components/YourTrips/CommuteGroup.tsx:
```diff
@@ -0,0 +1,114 @@
+import { useEffect, useState } from "react";
+import type { Key } from "react-aria-components";
```
Looks like this is not used anymore? We might be able to remove this import.

rachelekm replied:
Removed in latest rebase 👍

aaronxsu reviewed (approved):
Thanks for addressing the comments. I added two small questions, but they should not block merging. Good to go after a changelog entry.

rachelekm committed:
Update CHANGELOG

rachelekm commented:
Thanks so much @aaronxsu and @philippschmitt! Addressed those last comments, resolved merge conflicts on latest rebase and added a changelog entry. Will squash and merge on successful CI.
